pulpcore/pulpcore/app/serializers/base.py

```diff
@@ -217,6 +217,12 @@ def to_representation(self, instance):
 
         return ret
 
+    def validate(self, data):
+        unknown_fields = set(self.initial_data.keys()) - set(self.fields.keys())
+        if unknown_fields:
+            raise serializers.ValidationError("Unknown fields: {}".format(", ".join(unknown_fields)))
```

Comment:
I forgot to mention that this needs i18n. And maybe use a mapping:

```python
raise serializers.ValidationError({unknown_fields.pop(): _('Unexpected field')})
```

Reply:
Will do, won't `pop()` extract a single field only though?

Reply:
Ah yea. Here we go:

```
    def validate(self, data):
        unknown_fields = set(self.initial_data.keys()) - set(self.fields.keys())
        if unknown_fields:
            unknown_fields = {field: _('Unexpected field') for field in unknown_fields}
            raise serializers.ValidationError(unknown_fields)
        return data
```

Reply:
Perfect.